mitmproxy/optmanager.py

```diff
@@ -365,6 +357,20 @@ def parse_setval(self, o: _Option, optstr: typing.Optional[str], currentvalue: t
                     return [optstr]
         raise NotImplementedError("Unsupported option type: %s", o.typespec)
 
+    @staticmethod
+    def _parse_numeric(optstr: typing.Optional[str], name: str, typespec: type) -> typing.Union[int, float, None]:
+        """Helper method to parse a numeric option value from a string"""
+        if optstr:
+            try:
+                return typespec(optstr)
```

Comment:
This looks like it will fail for options of type `Optional[int]` or `Optional[float]`? 
Please add a test to convince me otherwise. 😄

Reply:
I've added tests on this commit: https://github.com/mitmproxy/mitmproxy/pull/4927/commits/310d4b5b3b2d63fe64a922b80a7295740a329fc9